### PR TITLE
feat/namespaced-discoveryservice

### PR DESCRIPTION
**Pull request by roivaz**
This PR is focused in enabling namespaced installation of marin3r. To do so the following changes have been made:

* The MutatingWebhook to inject envoy sidecars to Pods is now deployed during the operator installation. By doing so we free the DiscoveryService controller from doing this task itselfm which comes with the advantage that the DiscoveryService is not required to manage cluster scoped resources anymore
* Thanks to the previous point, the DiscoveryService can now be converted into a namespaced resource. By doing so, each namespace can has its own DiscoveryResource that only watches resources inside it. This is a breaking change and makes it imposible to have an upgrade path from previous versions of the operator. The only way to upgrade is to uninstall the previous version and the install the new v0.7.x release.
* The installation manifests have been updated to allow for two different install options for the operator:
  * Directly deploying the operator using kustomize. This option requires cert-manager to be present in the cluster in order to provision the required certificates for the mutating webhook.
  * Intall the operator through OLM. In this case OLM itself provisions autimatically the required certificates for the webhook and cert-manager is not required. 

Related to #58

### CODE REVIEW

**Review by raelga — APPROVED** [review]
Awesome job as always.

**Review by raelga — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -62,6 +64,10 @@ deploy-test: manifests kustomize
 undeploy-test: manifests kustomize
 	$(KUSTOMIZE) build config/test | kubectl delete -f -
 
+deploy-cert-manager:
+	# kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.1.0/cert-manager.crds.yaml
```
Comment (slopezz) [inline Makefile:68]: Comment?

**Review by slopezz — CHANGES_REQUESTED** [review]
Awesome change from cluster scope to namespace scope, security concerns are no longer a problem!

It is a very important change, and I see that documentation has not been updated yet.

Doing a fast review, at least (maybe more) the following docs needs to reflect that change (docs, yaml examples embedded into docs, network diagrams...) :
https://github.com/3scale/marin3r#installation
https://github.com/3scale/marin3r/blob/master/docs/design/discovery-service.md
https://github.com/3scale/marin3r/blob/master/docs/design/operator.md#discoveryservice-controller
https://github.com/3scale/marin3r/blob/master/docs/design/sidecar-injection.md#enable-sidecar-injection


Another thing to take into account for docs, is that now that `DiscoveryService` is namespace scope, certificates must live in the namespace, so `EnvoyConfigs` does not require to specify the namespace like until now, where you could have the cert on a different namespace and configure it on the EC:
```yaml
  envoyResources:
    secrets:
      - name: kuard-certificate
        ref:
          name: kuard-certificate
          namespace: default  ### This one
```

**Comment by roivaz** [comment]
Yes, docs/examples need an update, but I'll do that in a documentation specific PR.

Regarding the certificates that's true, Secrets must live in the same namespace, which is better security wise.

**Comment by slopezz** [comment]
> Yes, docs/examples need an update, but I'll do that in a documentation specific PR.
> 
> Regarding the certificates that's true, Secrets must live in the same namespace, which is better security wise.

Perfect!

File: Makefile (hunk 1 above)
Comment (roivaz) [reply Makefile]: Fixed!

**Review by slopezz — APPROVED** [review]

**Pull request merged**